pkg/kubectl/sorting_printer.go

```diff
@@ -60,8 +61,28 @@ func (s *SortingPrinter) sortObj(obj runtime.Object) error {
 	if len(objs) == 0 {
 		return nil
 	}
+
+	if err := SortObjects(objs, s.SortField); err != nil {
+		return err
+	}
+
+	switch list := obj.(type) {
+	case *v1.List:
+		for ix := range objs {
+			data, err := v1.Codec.Encode(objs[ix])
```

Comment:
Yeah, we can't do this.  List objects must remain opaque.

On Tue, Oct 20, 2015 at 11:12 AM, David Eads <notifications@github.com>
wrote:

> In pkg/kubectl/sorting_printer.go
> <https://github.com/kubernetes/kubernetes/pull/15928#discussion_r42508662>
> :
>
> > @@ -60,8 +61,28 @@ func (s *SortingPrinter) sortObj(obj runtime.Object) error {
> >  	if len(objs) == 0 {
> >  		return nil
> >  	}
> > +
> > +	if err := SortObjects(objs, s.SortField); err != nil {
> > +		return err
> > +	}
> > +
> > +	switch list := obj.(type) {
> > +	case *v1.List:
> > +		for ix := range objs {
> > +			data, err := v1.Codec.Encode(objs[ix])
>
> This is non-obvious to me. Lists can contain objects done in different
> codecs. Why force them out of their native shape?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/15928/files#r42508662>.
>